### PR TITLE
GG-35619 Fix node failures due to java.nio.channels.CancelledKeyException

### PR DESCRIPTION
**Pull request by rpuch**
https://ggsystems.atlassian.net/browse/GG-35619

### CODE REVIEW

**Review by ibessonov — COMMENTED** [review]
I can't validate your fix and assume that it's correct. My comments are only related to test code.

File: modules/core/src/test/java/org/apache/ignite/internal/util/nio/FastSessionMoveAndKeyCancellationTest.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+        try (
+            IgniteEx ignite = startGrid(1);
+            IgniteClient client1 = startClient()
+        ) {
+            IgniteInternalFuture<?> clientJobsFuture = multithreadedAsync(new ReconnectConstantly(), 2);
+
+            try {
+                assertFalse("CancelledKeyException was thrown", failureLogMessagesLatch.await(1, TimeUnit.MINUTES));
+            } finally {
```
[the remark this answers is not in the document]
Comment (rpuch) [reply modules/core/src/test/java/org/apache/ignite/internal/util/nio/FastSessionMoveAndKeyCancellationTest.java:98]: I did run checkstyle manually like this: `mvn clean install -DskipTests -Pcheckstyle`, checkstyle thinks everything is ok.

What exactly seems not right in this code?

File: modules/core/src/test/java/org/apache/ignite/internal/util/nio/FastSessionMoveAndKeyCancellationTest.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            while (!Thread.currentThread().isInterrupted()) {
+                try (IgniteClient ignored = startClient()) {
+                    // do nothing, just let the client disconnect
+                }
+            }
+
+            return null;
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (rpuch) [reply modules/core/src/test/java/org/apache/ignite/internal/util/nio/FastSessionMoveAndKeyCancellationTest.java]: Added a trailing new line

File: modules/core/src/test/java/org/apache/ignite/internal/util/nio/FastSessionMoveAndKeyCancellationTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (rpuch) [reply modules/core/src/test/java/org/apache/ignite/internal/util/nio/FastSessionMoveAndKeyCancellationTest.java:98]: Checkstyle in TC is green: https://ggtc.gridgain.com/buildConfiguration/GridGain8_Test_SanityCheck_Checkstyle?branch=gg-35619&mode=builds

File: modules/core/src/test/java/org/apache/ignite/internal/util/nio/FastSessionMoveAndKeyCancellationTest.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+     *
+     * This test makes sure this does not happen.
+     */
+    @SuppressWarnings("unused")
+    @Test
+    @WithSystemProperty(key = GridNioServer.IGNITE_IO_BALANCE_RANDOM_BALANCE, value = "true")
+    public void quickSessionMovingBetweenWorkersShouldNotTriggerCancelledKeyException() throws Exception {
+        CountDownLatch failureLogMessagesLatch = new CountDownLatch(2);
+
+        logger.registerListener(
```
[the remark this answers is not in the document]
Comment (rpuch) [reply modules/core/src/test/java/org/apache/ignite/internal/util/nio/FastSessionMoveAndKeyCancellationTest.java:81]: The code with a latch seems more straightforward to me

File: modules/core/src/test/java/org/apache/ignite/internal/util/nio/FastSessionMoveAndKeyCancellationTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ibessonov) [reply modules/core/src/test/java/org/apache/ignite/internal/util/nio/FastSessionMoveAndKeyCancellationTest.java:98]: It should be
```
}
finally {
```
It's weird that checkstyle is fine with this code, maybe it ignores tests or something, I don't know

**Review by ibessonov — APPROVED** [review]

File: modules/core/src/test/java/org/apache/ignite/internal/util/nio/FastSessionMoveAndKeyCancellationTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (rpuch) [reply modules/core/src/test/java/org/apache/ignite/internal/util/nio/FastSessionMoveAndKeyCancellationTest.java]: Oh... it was embarassing :) Fixed it

**Pull request merged**